src/openrct2/core/String.cpp

```diff
@@ -823,6 +826,30 @@ namespace String
 
         return trunc;
     }
+
+    std::string URLEncode(std::string_view value)
+    {
+        std::ostringstream escaped;
+        escaped.fill('0');
+        escaped << std::hex;
+
+        for (auto c : value)
+        {
+            // Keep alphanumeric and other accepted characters intact
+            if (std::isalnum(c) || c == '-' || c == '_' || c == '.' || c == '~')
+            {
+                escaped << c;
+                continue;
+            }
+
+            // Any other characters are percent-escaped
+            escaped << std::uppercase;
+            escaped << '%' << std::setw(2) << int32_t(static_cast<unsigned char>(c));
+            escaped << std::nouppercase;
```

Comment:
Maybe
```suggestion
            if (std::isalnum(c) || c == '-' || c == '_' || c == '.' || c == '~')
            {
                escaped << c;
            }
            else
            {
                // Any other characters are percent-escaped
                escaped << std::uppercase;
                escaped << '%' << std::setw(2) << int32_t(static_cast<unsigned char>(c));
                escaped << std::nouppercase;
            }
```
I see the benefits of using continue when the rest of the function is big, but in this case seems like we can go with `if...else` to be clearer.